### PR TITLE
st2client: Add limit/"-n" flag for key list

### PR DESCRIPTION
**Pull request by humblearner**
```
$ st2 key list -n 1
+------+-------+--------+-----------+--------------+------+------------------+
| name | value | secret | encrypted | scope        | user | expire_timestamp |
+------+-------+--------+-----------+--------------+------+------------------+
| a    | b     | False  | False     | st2kv.system |      |                  |
+------+-------+--------+-----------+--------------+------+------------------+
+----------------------------------------------------------------------------------+
| Note: Only one key value pair is displayed. Use -n/--last flag for more results. |
+----------------------------------------------------------------------------------+
```
```
$ st2 key list -n 3
+-------+-------------------------+--------+-----------+--------------+------+------------------+
| name  | value                   | secret | encrypted | scope        | user | expire_timestamp |
+-------+-------------------------+--------+-----------+--------------+------+------------------+
| a     | b                       | False  | False     | st2kv.system |      |                  |
| apple | 0083E1033A2A09636E7CF26 | True   | True      | st2kv.system |      |                  |
|       | F7E0008983A47B67307E151 |        |           |              |      |                  |
|       | 284FC0649EFDFCF30F2664C |        |           |              |      |                  |
|       | 67D525ADD1F387FE9B7CB76 |        |           |              |      |                  |
|       | 0A3151FEDFFC0FCF1EF528  |        |           |              |      |                  |
| key_  | value                   | False  | False     | st2kv.system |      |                  |
+-------+-------------------------+--------+-----------+--------------+------+------------------+
+-----------------------------------------------------------------------------------------------+
| Note: Only first 3 key value pairs are displayed. Use -n/--last flag for more results.        |
+-----------------------------------------------------------------------------------------------+

```

### CODE REVIEW

**Comment by Kami** [comment]
Let's please also add a changelog entry, unit test and some tests in st2tests repo :)

**Comment by humblearner** [comment]
robot tests: https://github.com/StackStorm/st2tests/pull/110

File: st2client/st2client/commands/keyvalue.py [hunk 1]
```diff
@@ -77,23 +80,40 @@ def __init__(self, *args, **kwargs):
...
 
-    def run_and_print(self, args, **kwargs):
+    @resource.add_auth_token_to_kwargs_from_cli
+    def run(self, args, **kwargs):
+        # Filtering options
         if args.prefix:
             kwargs['prefix'] = args.prefix
-
         decrypt = getattr(args, 'decrypt', False)
-        kwargs['params'] = {'decrypt': str(decrypt).lower()}
```
Comment (Kami) [inline st2client/st2client/commands/keyvalue.py]: Any particular reason you moved away from params?

I believe I explicitly used params to make it clear those are query param and not some other random arguments method takes.

Comment (Kami) [inline st2client/st2client/commands/keyvalue.py]: Besides that, LGTM.

Comment (humblearner) [inline st2client/st2client/commands/keyvalue.py]: This is what happens when params is used:
```
>>> kwargs={'token': u'c992ce735afd496fb9c17fa0499ce3a2', 'params': {'scope': 'system', 'decrypt': 'false', 'user': None}}
>>> token = kwargs.get('token', None)
  2 api_key = kwargs.get('api_key',None)
  3 params = kwargs.get('params', {})
>>> for k, v in six.iteritems(kwargs):
  2             # Note: That's a special case to support api_key and token kwargs
  3             if k not in ['token', 'api_key']:
  4                 params[k] = v
>>> params
{'scope': 'system', 'decrypt': 'false', 'user': None, 'params': {...}}
>>> urllib.parse.urlencode(params)
'scope=system&decrypt=false&user=None&params=%7B%27scope%27%3A+%27system%27%2C+%27decrypt%27%3A+%27false%27%2C+%27user%27%3A+None%2C+%27params%27%3A+%7B...%7D%7D'
>>>
```
here: https://github.com/StackStorm/st2/blob/master/st2client/st2client/models/core.py#L251

Comment (Kami) [inline st2client/st2client/commands/keyvalue.py]: Ah, that's a bit nasty, should perhaps update that loop which iterates over `kwargs` to skip param.

I would prefer that personally over "kwargs" abuse (with kwargs abuse you don't know what is a query param and what is a something else).

Comment (humblearner) [inline st2client/st2client/commands/keyvalue.py]: @kami: Added the params back, can you please skim through if the change is ok: https://github.com/StackStorm/st2/blob/add_st2client_keystore_limit_param/st2client/st2client/models/core.py#L257

**Pull request merged**